### PR TITLE
Project Entity and GET route

### LINKED ISSUE #14: Build repository data object
### Pre-requisites

* Understand GitHub API JSON response

### Describe the solution you'd like

* Create data type for repository information.
#### Basic repository data type
* id (int)
* owner (string) //(type githubAccount?)
* name (string)
* full_name (string)
* html_url (string)
* description (string)
* api_url (string)
* forks_url(string)
* forks_count (int)
* issues_url (string)
* open_issues_count (int)
* pulls_url (string)
* topics (list)
 
### Additional context

* May need githubAccount object

### PR DESCRIPTION
**Pull request by NotKevin**
### Pre-Requisites

- [x] Yes, I updated [Authors.md](../blob/main/Authors.md) **OR** this is not my first contribution
- [x] Yes, I included and/or modified tests to cover relevant code **OR** my change is non-technical
- [x] Yes, I wrote this code entirely myself **OR** I properly attributed these changes in [Third Party Notices](../blob/main/THIRD-PARTY-NOTICES.txt)

### Description of Changes

- "Project" Entity
- GET route to return repository data from GitHub API

### Related Issues

Closes #14

### CODE REVIEW

**Review by dre8597 — DISMISSED** [review]
Looking good so far, I would change the PR name to Project from Repository so others won't get confused 🙂

File: packages/api/src/api/project.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    try {
+      const projectList = await req.entityManager.find(Project, {});
+      
+      const repoNodeIds = projectList.map(project => project.nodeID);
+
+      const fetchRes = await fetch('https://api.github.com/graphql', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+          'Authorization': `bearer ${req.user?.githubToken}`,
```
Comment (NotKevin) [inline packages/api/src/api/project.ts:18]: Only problem I am running into during tests is testing branching on "req.user?.githubToken". Is there any way to mock the session?

Comment (johnkahn) [reply packages/api/src/api/project.ts:18]: On your test handler, you can pass a second middleware to modify the request. Should look like this

```typescript
const handler = await testHandler(yourHandler, (req, _res, next) => {
  req.user = { githubToken: 'token' };
  next();
});
```

File: packages/api/src/api/users.ts [hunk 2]
```diff
@@ -40,45 +40,47 @@ users.get('/:userId', async (req, res) => {
...
+
+      user.assign(sanitizedUser);
+
+      await req.entityManager.flush();
+      res.send(user);
+    } catch (error) {
+      logger.error(`There was an issue updating user "${userId}"`, error);
+      res.status(500).send(`There was an issue updating user "${userId}"`);
+    }
+  } else res.sendStatus(401);
```
Comment (johnkahn) [inline packages/api/src/api/users.ts:85]:
```suggestion
  } else {
    res.sendStatus(401);
  }
```

**Review by johnkahn — CHANGES_REQUESTED** [review]
Quick change on the else

**Comment by johnkahn** [comment]
Looks like there are some conflicts since we just merged your other PR #50. After those are resolved and the checks passed we can merge this one

**Review by johnkahn — APPROVED** [review]

**Pull request merged**